data/blog/session-replay-sdk-bundle-size-optimizations.md

```diff
@@ -0,0 +1,209 @@
+
+In order to achieve these bundle size improvements, we took a couple of steps ranging from removing unused code to build time configuration and improved tree shaking:
+
+- Allow to remove iframe & shadow DOM support via a build-time flag
+- Removed canvas recording support by default (users can opt-in via a config option, [support is coming](https://github.com/getsentry/sentry-javascript/issues/6519))
+- Remove unused code from our rrweb fork
+- Remove unused code in Session Replay itself
+- Allow to remove the included compression worker in favor of hosting it yourself
+- Move to a different compression library with a smaller footprint
+
+## Primer: rrweb
+
+[rrweb](https://github.com/getsentry/rrweb) is the underlying tool we use to make the recordings for Session Replay.
+While we try to contribute to the main rrweb repository as much as possible, there are some changes that are very specific to our needs at Sentry,
+which is why we also maintain a [forked version](https://github.com/getsentry/rrweb) of rrweb with some custom changes.
+
+## Primer: Tree Shaking
+
+If you do not know what tree shaking is and how it works, you can [read about it in our docs](https://docs.sentry.io/platforms/javascript/configuration/tree-shaking/).
+
+## Allow to remove iframe & shadow DOM support via a build-time flag
+
+While rrweb allows to capture more or less everything that happens on your page, some of the things it can capture may not be necessary for some users.
+For these cases, we now allow users to manually remove certain parts of the rrweb codebase they may not need at build time, reducing the bundle size.
+
+In [getsentry/sentry-javascript#9274](https://github.com/getsentry/sentry-javascript/pull/9274) & [getsentry/rrweb#114](https://github.com/getsentry/rrweb/pull/114) we implemented the ground work to allow to tree shake iframe and shadow dom recording. This means that if, for example, you don't have any iframes on your page, you can safely opt-in to remove this code from your application.
+
+In [getsentry/sentry-javascript-bundler-plugins#428](https://github.com/getsentry/sentry-javascript-bundler-plugins/pull/428) we implemented an easy way to implement these optimizations in your app. If you are using one of our bundler plugins:
+
+- [@sentry/webpack-plugin](https://www.npmjs.com/package/@sentry/webpack-plugin)
+- [@sentry/vite-plugin](https://www.npmjs.com/package/@sentry/vite-plugin)
+- [@sentry/rollup-plugin](https://www.npmjs.com/package/@sentry/rollup-plugin)
+- [@sentry/esbuild-plugin](https://www.npmjs.com/package/@sentry/esbuild-plugin)
+
+You can just update to its latest version, and add this configuration to the plugin:
+
+```js
+sentryPlugin({
+  bundleSizeOptimizations: {
+    excludeDebugStatements: true,
+    excludeReplayIframe: true,
+    excludeReplayShadowDom: true,
+  },
+})
+```
+
+This will safe you about 5 KB gzipped of bundle size!
+
+### How we implement build-time tree shaking flags
```

Comment:
```suggestion
### How we implemented build-time tree shaking flags
```

Reply:
Okay, thinking on this -- in the headings we haven't been consistent about using present or past tense.

For example, we say "Remove*d* canvas recording support by default" but in other places say "Update library used for compression" -- I think we just need to pick one way or the other and stick w/ it for all the headings

Reply:
udpated it to use past tense everywhere!